src/browserlib/extract-algorithms.mjs

```diff
@@ -613,9 +613,20 @@ function findAlgorithms(section, { includeIgnored } = { includeIgnored: false })
     })
     .filter(algo => includeIgnored || !!algo.rationale);
 
+  // Probable one-step algorithms starts with "To "
+  // followed by an exported definition
+  // of dfn-type either "dfn" or "abstract-op"
+  const candidateDfnSelectors = ['dfn[data-export][data-dfn-type="dfn"]',
+				 'dfn[data-export][data-dfn-type="abstract-op"]'
+				];
+  const probableOneLine = [...section.querySelectorAll(candidateDfnSelectors.map(s => `p:has(${s})`).join(','))]
+    .filter(p => p.textContent.startsWith("To " + p.querySelector(candidateDfnSelectors.join(',')).textContent))
+    .map(p => {
+      return { rationale: '"To <dfn>"', root: p };
```

Comment:
Isn't that going to extract the introductory paragraph of multi-step algorithms too, which will also be extracted with an `<ol>` as root?

```html
<p>To <dfn ...>do plenty of things</dfn>, run the following steps:</p>
<ol><li>...</li></ol>
```

The duplicate filtering some lines below won't catch that because the `<p>` and `<ol>` elements are disjoint.

Reply:
I had (mistakenly) assumed that the duplication filter would handle it (or that the test would yell at me otherwise); will look into fixing it

Reply:
done in [ee10949](https://github.com/w3c/reffy/pull/1614/commits/ee109493f3fdb8d1c7b82200f5d31350d7d20027)